### PR TITLE
fix(ME-2316): only send windowschange if ssm is connected

### PR DESCRIPTION
**Pull request by btoonk**
# Description

only send windowschange if ssm is connected
Fixes # (ME-2316)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Tested with local build

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

File: internal/ssh/session/ssm_session.go [hunk 1]
```diff
@@ -639,6 +639,10 @@ func (s *ssmSession) OpenDataChannel(log log.T) (err error) {
 }
 
 func (s *ssmSession) handleWindowChange(width, height int) error {
+	if s.DataChannel == nil {
```
Comment (adrianosela) [inline internal/ssh/session/ssm_session.go:642]: Should DataChannel have a lock around it?

**Review by adrianosela — APPROVED** [review]
Comment about locking

File: internal/ssh/session/ssm_session.go (hunk 1 above)
Comment (btoonk) [reply internal/ssh/session/ssm_session.go:642]: No I think we are fine.

Comment (adrianosela) [reply internal/ssh/session/ssm_session.go:642]: What about enqueue the requests until we can use them instead of discarding them?

Comment (btoonk) [reply internal/ssh/session/ssm_session.go:642]: We already are storing the values, and use those values after we initialise the ssm session.:
https://github.com/borderzero/border0-cli/blob/61713d48f6bda37e07b2a87b9157049e894601a8/internal/ssh/session/ssm_session.go#L532

**Pull request merged**